synapse/lib/ast.py

```diff
@@ -1299,50 +1300,82 @@ class TagCond(Cond):
     #foo.bar
     '''
     def getLiftHints(self):
-        name = self.kids[0].value()
+
+        kid = self.kids[0]
+
+        if not isinstance(kid, TagMatch):
+            # TODO:  we might hint based on variable value
+            return ()
+
+        name = kid.value()
         if '*' in name:
             return ()
         return (
             ('tag', {'name': name}),
         )
 
-    def getCondEval(self, runt):
+    async def getCondEval(self, runt):
 
-        name = self.kids[0].value()
+        assert len(self.kids) == 1
+        kid = self.kids[0]
+        if isinstance(kid, TagMatch):
+            name = self.kids[0].value()
+        else:
+            # TODO:  enable runtval calc here (variable nodes haven't been evaluated yet)
+            # if kid.isRuntSafe(runt):
+            #     name = await kid.runtval(runt)
+            name = None
+
+        if name is not None:
+
+            # Allow for a user to ask for #* to signify "any tags on this node"
+            if name == '*':
+                async def cond(node, path):
+                    # Check if the tags dictionary has any members
```

Comment:
this could also be `return bool(node.tags)`

Reply:
done